smart-contract-usage-in-javascript-sdk.md

```diff
@@ -0,0 +1,330 @@
+
+## Prerequisites
+
+ - up to date npm and nodejs installed
+ - private and public key for an account with some balance ([create an account](account-creation-in-ae-cli.md); [get testnet tokens](get-testnet-tokens.md))
+
+## Setup
+
+Create a project directory
+
+```
+mkdir aepp-smart-contract-usage-tutorial
+cd aepp-smart-contract-usage-tutorial
+```
+
+## Define Example Contract
+
+We define an example contract which features functions for reserving something if a minimum price is payed to the contract. The contract creator can later withdraw the payed tokens. Each line is commented with what it does. Sophia language definition can be found in [the protocol repository](https://github.com/aeternity/protocol/blob/master/contracts/sophia.md).
+
+Save this as `ExampleContract.aes`.
+
+```
+// contract definition
+contract ExampleContract =
+
+
+  // state definition, with available count, set minimum price and the owner
+  record state = { available_count : int, min_price : int, owner : address }
+
+
+  // defining the require function later used to check calling constraints
+  private function require(b : bool, err : string) =
+    if(!b) abort(err)
+
+
+  // function to initialize the contract, setting the available count, minimum price and the owner
+  public function init(available_count' : int, min_price' : int) : state =
+    { available_count = available_count', min_price = min_price', owner = Call.caller }
+
+
+  // function for anyone to reserve some count requiring available count and the minimum price payed
+  public stateful function reserve(count: int) =
+  
+    // requiring the price payed per item to be above or equal the set minimum price
+    require(Call.value >= state.min_price * count, "Payed price is less than minimum required")
+    let remaining_count = state.available_count - count
+    
+    // requiring count to be available
+    require(remaining_count >= 0, "Less available then to be reserved")
+    
+    // applying remaining count to the state
+    put(state{ available_count = remaining_count })
+
+
+  // function to get the remaining count, can be statically called as it doesn't change state
+  public function available_count() : int =
+    state.available_count
+
+
+  // function only to be called by the owner to withdraw all balance that was sent to the contract
+  public stateful function owner_withdraw() : int=
+  
+    // requiring contract caller to be the registered owner
+    require(Call.caller == state.owner, "Only owner can withdraw")
+    let withdrawed_balance = Contract.balance
+    
+    // creating spend of the balance to the registered owner
+    Chain.spend(state.owner, Contract.balance)
+    
+    // returning the balance that was withdrawed
+    withdrawed_balance
+```
+
+## Nodejs Environment
+
+1. Create a directory for the project and initialize it with npm `npm init -y`
```

Comment:
I think that we do not need to create a new directory as we have done so in the setup phase